### PR TITLE
experiment: add defsym map

### PR DESCRIPTION
**Pull request by mlutze**

### CODE REVIEW

**Comment by mlutze** [comment]
@magnus-madsen is this roughtly what you're looking for?

File: main/src/ca/uwaterloo/flix/language/ast/TypedAst.scala [hunk 1]
```diff
@@ -33,7 +33,9 @@ object TypedAst {
                   typeAliases: Map[Symbol.TypeAliasSym, TypedAst.TypeAlias],
                   entryPoint: Option[Symbol.DefnSym],
                   sources: Map[Source, SourceLocation],
-                  classEnv: Map[Symbol.ClassSym, Ast.ClassContext])
+                  classEnv: Map[Symbol.ClassSym, Ast.ClassContext],
+                  defSyms: Map[List[String], Symbol.DefnSym]
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/ast/TypedAst.scala:37]: I would put it first, and into one data structure  (i.e. one data structure for both defs and enums and ...)

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/ast/TypedAst.scala:37]: I also would like some Ast.Namespace abstraction. (Perhaps we should do that first?)

**Comment by magnus-madsen** [comment]
Can you update this to reflect https://github.com/flix/flix/issues/4890

Also, I would like these new fields to be first in TypedAst.Root

**Comment by magnus-madsen** [comment]
Or maybe `modules` is first and `names` is last.

**Pull request closed without merging**